x-pack/packages/security/plugin_types_public/src/authorization/authorization_service.ts

```diff
@@ -10,6 +10,17 @@ export interface AuthorizationServiceSetup {
    * Determines if role management is enabled.
    */
   isRoleManagementEnabled: () => boolean | undefined;
+
+  /**
+   * Retrieve current user's API key privileges.
+   */
+  getCurrentUserApiKeyPrivileges: () => Promise<AuthorizationCurrentUserApiKeyPrivilegesResponse>;
+}
+
+export interface AuthorizationCurrentUserApiKeyPrivilegesResponse {
```

Comment:
Can't we use `GetAPIKeysResult` interface here?

Reply:
This code is in a package, but `GetAPIKeysResult` is the server-side plugin. I'm thinking this would create a circular dependency problem, no? The `GetAPIKeysResult` would need to be moved to shared package to be reused?

Reply:
Thanks for the ping! I added my feedback [here](https://github.com/elastic/kibana/pull/183286#discussion_r1605158966)